src/lib.rs

```diff
@@ -812,447 +825,453 @@ impl Default for Section {
-    let l2 = &l1 + 10;
-    assert_eq!(l2.offset(&l1).unwrap(), 10);
-}
+    #[test]
+    fn label_add() {
+        let l1 = Label::new();
+        let l2 = &l1 + 10;
+        assert_eq!(l2.offset(&l1).unwrap(), 10);
+    }
 
-#[test]
-fn label_sub() {
-    let l1 = Label::new();
-    let l2 = &l1 - 10;
-    assert_eq!(l2.offset(&l1).unwrap(), -10);
-}
+    #[test]
+    fn label_sub() {
+        let l1 = Label::new();
+        let l2 = &l1 - 10;
+        assert_eq!(l2.offset(&l1).unwrap(), -10);
+    }
 
-#[test]
-fn label_sub_label() {
-    let l1 = Label::new();
-    let l2 = &l1 + 10;
-    assert_eq!(&l2 - &l1, 10);
-}
+    #[test]
+    fn label_sub_label() {
+        let l1 = Label::new();
+        let l2 = &l1 + 10;
+        assert_eq!(&l2 - &l1, 10);
+    }
 
-#[test]
-fn label_set_const() {
-    let l = Label::new();
-    let val = 0x12345678;
-    l.set_const(val);
-    assert_eq!(l.value().unwrap(), val);
-}
+    #[test]
+    fn label_set_const() {
+        let l = Label::new();
+        let val = 0x12345678;
+        l.set_const(val);
+        assert_eq!(l.value().unwrap(), val);
+    }
 
-#[test]
-fn label_set() {
-    let val = 0x12345678;
-    let l1 = Label::from_const(val);
-    let l2 = Label::new();
-    l2.set(&l1);
-    assert_eq!(l2.value().unwrap(), val);
-    // Check that setting the first value's label *after* the call to set works.
-    let l3 = Label::new();
-    let l4 = Label::new();
-    l4.set(&l3);
-    l3.set_const(val);
-    assert_eq!(l4.value().unwrap(), val);
-}
+    #[test]
+    fn label_set() {
+        let val = 0x12345678;
+        let l1 = Label::from_const(val);
+        let l2 = Label::new();
+        l2.set(&l1);
+        assert_eq!(l2.value().unwrap(), val);
+        // Check that setting the first value's label *after* the call to set works.
+        let l3 = Label::new();
+        let l4 = Label::new();
+        l4.set(&l3);
+        l3.set_const(val);
+        assert_eq!(l4.value().unwrap(), val);
+    }
 
-#[test]
-fn section_construction() {
-    let s = Section::new();
-    assert_eq!(s.endian, DEFAULT_ENDIAN);
+    #[test]
+    fn section_construction() {
+        let s = Section::new();
+        assert_eq!(s.endian, DEFAULT_ENDIAN);
 
-    let s2 = Section::with_endian(Endian::Little);
-    assert_eq!(s2.endian, Endian::Little);
-}
+        let s2 = Section::with_endian(Endian::Little);
+        assert_eq!(s2.endian, Endian::Little);
+    }
 
-#[test]
-fn section_append_bytes() {
-    let s = Section::new();
-    let b1 = [0, 1, 2, 3, 4];
-    let b2 = [0xf, 0xe, 0xd, 0xc, 0xb];
-    assert_eq!(
-        s.append_bytes(&b1)
-            .append_bytes(&b2)
-            .get_contents()
-            .unwrap(),
-        &[0, 1, 2, 3, 4, 0xf, 0xe, 0xd, 0xc, 0xb]
-    );
-}
+    #[test]
+    fn section_append_bytes() {
+        let b1 = [0, 1, 2, 3, 4];
+        let b2 = [0xf, 0xe, 0xd, 0xc, 0xb];
+        assert_eq!(
+            Section::inline(None, |s| s.append_bytes(&b1).append_bytes(&b2))
```

Comment:
This makes me wonder if the API the tests want isn't actually: `fn Section::contents_with( /* the same args you have */ ) -> Vec<u8>`.

I don't think it matters much either way.

Reply:
Yah, could be a second pass, I just did the simpler things for now I think.